packages/js/core/src/types/WrapError.ts

```diff
@@ -0,0 +1,343 @@
+import { CleanResolutionStep } from "../algorithms";
+
+/**
+Wrap error codes provide additional context to WrapErrors.
+
+Error code naming convention (approximate):
+    type of handler
+    type of functionality
+    piece of functionality
+    ==> handler_typeFn_pieceFn
+
+Error code map:
+  0-24 -> Client
+  25-49 -> URI resolution
+  50-74 -> Wasm wrapper invocation & sub-invocation
+  75-99 -> Plugin invocation & sub-invocation
+  100-255 -> Unallocated
+ */
+export enum WrapErrorCode {
+  UNKNOWN,
+  CLIENT_LOAD_WRAPPER,
+  CLIENT_GET_FILE,
+  CLIENT_GET_IMPLEMENTATIONS,
+  CLIENT_VALIDATE_RESOLUTION,
+  CLIENT_VALIDATE_ABI,
+  CLIENT_VALIDATE_RECURSIVE,
+  CLIENT_QUERY_MALFORMED,
```

Comment:
Can we remove these `CLIENT_QUERY_...` error codes, since I think we should remove the GraphQL dependency (and the `.query(...)` method) from the client entirely in the near future?

Reply:
yeah, for sure